packages/connectivity/src/scp-cf/destination/destination-cache.ts

```diff
@@ -140,16 +146,23 @@ function cacheRetrievedDestination(
   const expirationTime = expiresIn
     ? Date.now() + parseInt(expiresIn) * 1000
     : undefined;
-  cache.set(key, destination, expirationTime);
+  cache.set(key, { entry: destination, expires: expirationTime });
 }
 
 /**
- * @internal
+ * Set the custom cache instance.
+ * @param cache - Cache instance.
  */
-export const destinationCache = DestinationCache(
-  new Cache<Destination>({ hours: 0, minutes: 5, seconds: 0 })
-);
+export function setDestinationCache<T extends CacheInterface<Destination>>(
```

Comment:
[req] We should use the more strict `DestinationInterface` everywhere here so that we are sure that we are compatible with the API contract we communicate to customers.